pkg/service-manager/handlers.go

```diff
@@ -0,0 +1,281 @@
+			})
+	}
+	m := entities.ServiceInstanceEntityToModel(&si, &b)
+	return serviceinstance.NewGetServiceInstanceByNameOK().WithPayload(m)
+}
+
+func (h *Handlers) getServiceInstances(params serviceinstance.GetServiceInstancesParams, principal interface{}) middleware.Responder {
+	defer trace.Trace("")()
+	var services []*entities.ServiceInstance
+
+	var err error
+	opts := entitystore.Options{
+		Filter: entitystore.FilterExists(),
+	}
+	opts.Filter, err = utils.ParseTags(opts.Filter, params.Tags)
+	if err != nil {
+		log.Errorf(err.Error())
+		return serviceinstance.NewGetServiceInstancesBadRequest().WithPayload(
+			&models.Error{
+				Code:    http.StatusBadRequest,
+				Message: swag.String(err.Error()),
+			})
+	}
+
+	err = h.Store.List(flags.ServiceManagerFlags.OrgID, opts, &services)
+	if err != nil {
+		log.Errorf("store error when listing service instances: %+v", err)
+		return serviceinstance.NewGetServiceInstancesDefault(http.StatusInternalServerError).WithPayload(
+			&models.Error{
+				Code:    http.StatusInternalServerError,
+				Message: swag.String("internal server error while listing service instances"),
+			})
+	}
+	var bindings []*entities.ServiceBinding
+	err = h.Store.List(flags.ServiceManagerFlags.OrgID, opts, &bindings)
+	if err != nil {
+		log.Errorf("store error when listing service bindings: %+v", err)
+		return serviceinstance.NewGetServiceInstancesDefault(http.StatusInternalServerError).WithPayload(
+			&models.Error{
+				Code:    http.StatusInternalServerError,
+				Message: swag.String("internal server error while listing service bindings"),
+			})
+	}
+	bindingsMap := make(map[string]*entities.ServiceBinding)
+	for _, binding := range bindings {
+		bindingsMap[binding.Name] = binding
+	}
+	var serviceModels []*models.ServiceInstance
+	for _, service := range services {
+		binding := bindingsMap[service.Name]
+		serviceModels = append(serviceModels, entities.ServiceInstanceEntityToModel(service, binding))
+	}
+	return serviceinstance.NewGetServiceInstancesOK().WithPayload(serviceModels)
+}
+
+func (h *Handlers) deleteServiceInstanceByName(params serviceinstance.DeleteServiceInstanceByNameParams, principal interface{}) middleware.Responder {
+	defer trace.Trace("")()
+	e := entities.ServiceInstance{}
+
+	var err error
+	opts := entitystore.Options{
+		Filter: entitystore.FilterExists(),
+	}
+	err = h.Store.Get(flags.ServiceManagerFlags.OrgID, params.ServiceInstanceName, opts, &e)
+	if err != nil {
+		return serviceinstance.NewDeleteServiceInstanceByNameNotFound().WithPayload(
+			&models.Error{
+				Code:    http.StatusNotFound,
+				Message: swag.String("service instance not found"),
+			})
+	}
+	err = h.Store.Delete(flags.ServiceManagerFlags.OrgID, params.ServiceInstanceName, &entities.ServiceInstance{})
+	if err != nil {
+		return serviceinstance.NewDeleteServiceInstanceByNameNotFound().WithPayload(
+			&models.Error{
+				Code:    http.StatusNotFound,
```

Comment:
Would this really be a not found situation? I imagine this would only happen as a race condition of trying to delete the same instance twice concurrently